core/controllers/classifier.py

```diff
@@ -79,7 +79,10 @@ def verify_signature(message, vm_id, received_signature):
     if secret is None:
         return False
 
-    generated_signature = generate_signature(secret, message)
+    if message is None:
```

Comment:
There is still one problem, though. I didn't notice it earlier but if we go with `message` then `signature` will be generated from `message` part but since that is empty, `signature` is also going to be empty and we don't have the security now. We need something in `message`.